base_iface: Add driver

mkowalski opened the pull request:
This commits adds a `driver` field to the base interface. It is meant to
provide an information about the driver used by the specific interface
if available. As some interfaces do not have any driver assigned (e.g.
loopback), this field will not be always present.

Example output:

```yml
interfaces:
- name: ens3
  type: ethernet
  driver: virtio_net
  state: up
```

Since our CI has not physical NIC, we cannot have this new property
tested in CI. Only manually tested.

mkowalski commented:
Hey @cathay4t, I would appreciate a review. This is "draft" until we release nispor with the new field, but apart from this it's done

cathay4t reviewed (dismissed):
I have release nispor 1.2.18.

Even with `Cargo.toml` changes to use latest nispor, the compilation will fail.

Meanwhile, please split the changes of `cli/persist_nic.rs` to new commit.

mkowalski committed:
base_iface: Add driver property

This commits adds a `driver` field to the base interface. It is meant to
provide an information about the driver used by the specific interface
if available. As some interfaces do not have any driver assigned (e.g.
loopback), this field will not be always present.

Example output:

```yml
interfaces:
- name: ens3
  type: ethernet
  driver: virtio_net
  state: up
```

Since our CI has not physical NIC, we cannot have this new property
tested in CI. Only manually tested.

Signed-off-by: Mat Kowalski <mko@redhat.com>
Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t committed:
ncl: Include driver information for `persist-nic-names` subcommand

As https://issues.redhat.com/browse/OCPBUGS-30256 indicates, there might
be network setup containing multiple NIC holding the same MAC address,
including driver name to systemd link file could solve their problem.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
The failure of CI on fedora is caused by Fedora 39 does not have latest rust-nispor yet. It will take a week or more for it to land there.

cathay4t commented:
Sorry I have to step in due to urgent request from stake holders.

cathay4t commented:
Tested in openshift upgrade:

```bash
sh-5.1# grep .* -H /etc/systemd/network/*.link
grep: ..: Is a directory
/etc/systemd/network/98-nmstate-eth0.link:# Generated by nmstate
/etc/systemd/network/98-nmstate-eth0.link:[Match]
/etc/systemd/network/98-nmstate-eth0.link:MACAddress=60:45:bd:49:eb:71
/etc/systemd/network/98-nmstate-eth0.link:Driver=hv_netvsc
/etc/systemd/network/98-nmstate-eth0.link:[Link]
/etc/systemd/network/98-nmstate-eth0.link:Name=eth0
/etc/systemd/network/98-nmstate-eth1.link:# Generated by nmstate
/etc/systemd/network/98-nmstate-eth1.link:[Match]
/etc/systemd/network/98-nmstate-eth1.link:MACAddress=60:45:bd:49:eb:71
/etc/systemd/network/98-nmstate-eth1.link:Driver=mlx4_core
/etc/systemd/network/98-nmstate-eth1.link:[Link]
/etc/systemd/network/98-nmstate-eth1.link:Name=eth1
/etc/systemd/network/98-nmstate-genev_sys_6081.link:# Generated by nmstate
/etc/systemd/network/98-nmstate-genev_sys_6081.link:[Match]
/etc/systemd/network/98-nmstate-genev_sys_6081.link:MACAddress=AE:01:7D:6B:04:60
/etc/systemd/network/98-nmstate-genev_sys_6081.link:[Link]
/etc/systemd/network/98-nmstate-genev_sys_6081.link:Name=genev_sys_6081
```

ffmancera reviewed (approved):
lgtm, i tested it on my system.
